content/blog/pulumi-in-a-cloud-native-world/index.md

```diff
@@ -0,0 +1,500 @@
+                "certData": "{{ cert }}",
+                "keyData": "{{ key }}"
+              }
+            }
+```
+
+## Putting all the Pieces Together
+
+As we have seen, each component of the `KEBAP` stack plays a crucial role in the overall architecture. Let's how the
+different workflows look like when we put all the pieces together. We go through the following scenarios:
+
+- Ignite the Control Plane Cluster With Pulumi
+- Ordering a New Development Kubernetes Cluster
+- Ordering a New Production Kubernetes Cluster
+- Ordering Other Infrastructure
+
+#### Ignite the Control Plane Cluster With Pulumi
+
+<img src="ignite.png" width="500px">
+
+We use Pulumi code to create the control plane cluster in our cloud provider of choice. After the creation of the
+control plane, we install the Argo CD in a very bare-bones way:
+
+```typescript
+const argocd = new ArgoCD("argocd", {
+    initialObjects: initialObjects
+}, {
+    providers: {
+        kubernetes: k8sProvider,
+    },
+});
+```
+
+The idea is to let then Argo CD fetch the rest of the configuration from the Git repository and apply it to the cluster.
+And here comes the power of GitOps: It will also manage the Argo CD itself. This way, we separated the workload running
+on the cluster from the creation of the cluster itself.
+
+#### Ordering a New Development Kubernetes Cluster
+
+<img src="order.png" width="500px">
+
+We use Backstage to order a new development Kubernetes cluster. As we want to keep the costs low for development
+cluster, we opted here for the [vCluster](https://www.vcluster.com/) project. The vCluster project allows us to create a
+Kubernetes cluster on demand, which runs as containers on the control plane cluster. The moment the cluster is created,
+we add it to the Argo CD as spoke cluster. This way, we can roll out applications to the development cluster in the same
+way as we would do for production clusters.
+
+See the definition of the `ClusterPolicy` object in the [Kyverno](#bonus-kyverno) section.
+
+#### Ordering a New Production Kubernetes Cluster
+
+<img src="order_cluster.png" width="500px">
+
+We use Backstage to order a new production Kubernetes cluster. This time, we use the Pulumi code to create the cluster
+in the cloud provider of choice. After the creation of the cluster, we add it to the Argo CD as a production cluster.
+This way, we can roll out applications to the production cluster in the same way as we would do for development
+clusters.
+
+#### Ordering Other Infrastructure
+
+<img src="order_rest.png" width="500px">
+
+We use Backstage to order other infrastructure components, such as databases, message queues, and storage. We use again
+the Pulumi Kubernetes Operator to create the infrastructure components in the Kubernetes cluster. This way, we can
+manage all the infrastructure components in the same way as we manage the Kubernetes clusters.
+
+## Conclusion and Pulumi Cloud as an Off-the-Shelf Solution
```

Comment:
what i don't agree here is Pulumi Cloud can be used in the KEBAP stack. I think you can just call this section the Conclusion